rpcserver.go

```diff
@@ -5243,3 +5256,76 @@ func (r *rpcServer) ChannelAcceptor(stream lnrpc.Lightning_ChannelAcceptorServer
 		}
 	}
 }
+
+// NewMacaroon allows the creation of a new macaroon with custom read and write
+// permissions. No first-party caveats are added since this can be done offline.
+func (r *rpcServer) NewMacaroon(ctx context.Context,
+	req *lnrpc.NewMacaroonRequest) (*lnrpc.NewMacaroonResponse, error) {
+
+	rpcsLog.Debugf("[newmacaroon]")
+
+	var (
```

Comment:
Could define these lists globally so that they aren't allocated every time. They can then also be close to the permission lists in this file.